subsys/shell/shell_fprintf.c

```diff
@@ -16,8 +17,14 @@ extern int _prf(int (*func)(), void *dest, char *format, va_list vargs);
 static int out_func(int c, void *ctx)
 {
 	const struct shell_fprintf *sh_fprintf;
+	const struct shell *shell;
 
 	sh_fprintf = (const struct shell_fprintf *)ctx;
+	shell = (const struct shell *)sh_fprintf->user_ctx;
+
+	if ((shell->shell_flag == SHELL_FLAG_OLF_CRLF) && (c == '\n')) {
```

Comment:
If the output string has a ``\r\n`` in it it will be changed to ``\r\r\n`` ...

Reply:
I do not see big issue with this. If this option is set, the shell typically sees only \n anyway, so there would be no \r there (usually). Having extra \r should not cause any issue as either as that would just mean that the cursor is returned to the start of the line.